huaweicloud/services/obs/resource_huaweicloud_obs_bucket_replication.go

```diff
@@ -0,0 +1,299 @@
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	obsClient, err := cfg.ObjectStorageClientWithSignature(region)
+	if err != nil {
+		return diag.Errorf("error creating OBS Client: %s", err)
+	}
+
+	bucket := d.Get("bucket").(string)
+	replicationRules, err := buildReplicationRules(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	opts := &obs.SetBucketReplicationInput{
+		Bucket: bucket,
+		BucketReplicationConfiguration: obs.BucketReplicationConfiguration{
+			Agency:           d.Get("agency").(string),
+			ReplicationRules: replicationRules,
+		},
+	}
+	_, err = obsClient.SetBucketReplication(opts)
+	if err != nil {
+		return diag.FromErr(getObsError("Error creating cross-region replication of OBS bucket", bucket, err))
+	}
+
+	// Assign the source bucket name as the resource ID
+	d.SetId(bucket)
+	return resourceObsBucketReplicationRead(ctx, d, meta)
+}
+
+func flattenDestinationBucket(output *obs.GetBucketReplicationOutput) string {
+	rules := output.ReplicationRules
+	if len(rules) == 0 {
+		return ""
+	}
+	return rules[0].DestinationBucket
+}
+
+func convertDestinationStatus(status obs.RuleStatusType) bool {
+	if status == obs.RuleStatusEnabled {
+		return true
+	}
+	return false
+}
+
+func convertDestinationHistoryObjectReplication(historyObjectReplication obs.EnabledType) bool {
+	if historyObjectReplication == obs.Enabled {
+		return true
+	}
+	return false
+}
+
+func flattenDestinationRules(output *obs.GetBucketReplicationOutput) []map[string]interface{} {
+	rules := output.ReplicationRules
+	if len(rules) == 0 {
+		return nil
+	}
+	result := make([]map[string]interface{}, len(rules))
+	for i, rule := range rules {
+		result[i] = map[string]interface{}{
+			"prefix":          rule.Prefix,
+			"storage_class":   rule.StorageClass,
+			"enabled":         convertDestinationStatus(rule.Status),
+			"history_enabled": convertDestinationHistoryObjectReplication(rule.HistoricalObjectReplication),
+			"id":              rule.ID,
+		}
+	}
+	return result
+}
+
+func resourceObsBucketReplicationRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	obsClient, err := cfg.ObjectStorageClientWithSignature(region)
+	if err != nil {
+		return diag.Errorf("error creating OBS Client: %s", err)
+	}
+
+	output, err := obsClient.GetBucketReplication(d.Id())
+	if err != nil {
+		if obsError, ok := err.(obs.ObsError); ok && obsError.Code == "ReplicationConfigurationNotFoundError" {
+			// The bucket does not have replication configurations
+			d.SetId("")
+			mErr := multierror.Append(nil,
+				d.Set("bucket", ""),
+				d.Set("destination_bucket", ""),
+				d.Set("agency", ""),
+				d.Set("rule", nil),
+			)
+			if err := mErr.ErrorOrNil(); err != nil {
+				return diag.Errorf("error setting OBS bucket replication fields: %s", err)
+			}
```

Comment:
the resource will no longer exist when you set ID to "", so don't need to care other fields
```suggestion
```